### PR TITLE
EECO-1486 Use Addrinfo#getaddrinfo instead of Resolv

### PR DESCRIPTION
**Pull request by capripot**
https://checkr.atlassian.net/browse/EECO-1486

### CODE REVIEW

File: lib/validate_url.rb [hunk 1]
```diff
@@ -72,16 +72,19 @@ def blacklisted_domain?(hostname)
...
-      def self.local?(hostname)
-        ip = begin
-               Resolv.getaddress(hostname)
-             rescue Resolv::ResolvError
-               nil
-             end
+      def self.local?(uri)
+        addr = begin
+                 Addrinfo.getaddrinfo(uri.hostname, get_port(uri), nil, :STREAM).sample
+               rescue SocketError
```
Comment (pranav-bagree) [inline lib/validate_url.rb:78]: getaddrinfo returns a SocketError when the DNS resolution fails. Should we return true in this case?

**Review by pranav-bagree — COMMENTED** [review]
We should update or add a test in checkr/webhooks to check for encoded internal URLs. https://github.com/checkr/webhooks/blob/b17c102732c578e88b2a1247be046e58dbac9aa8/spec/workers/webhook_processor_spec.rb#L163

File: lib/validate_url.rb (hunk 1 above)
Comment (capripot) [reply lib/validate_url.rb:78]: It would return a empty array if it doesnt resolve to anything. SocketError is when there is no DNS reachable I believe.

**Review by pranav-bagree — APPROVED** [review]

**Comment by capripot** [comment]
Addrinfo is depending on the machine it's running on for resolving DNS, so the example we got that would resolve to 169.xxx, it would happen only AWS server. Not sure if mocking the answer would be very useful.

**Pull request merged**